modules/python-modules/README.md

```diff
@@ -0,0 +1,648 @@
+
+```
+
+Acknowledgement mechanisms (ConsecutiveAckTracker, BatchedAckTracker) can be
+used similarly to how it was described at LogFetcher.
+
+## Making it more native
+
+All the examples above used some form of python() driver to be actually
+used, for instance, this was our Python based destination driver:
+
+```
+destination whatever {
+    python(class(mydestination.MyDestination)
+           options(option1 => value,
+                   option2 => value));
+};
+```
+
+while this works, this syntax is a bit foreign from a syslog-ng perspective
+and somewhat hard to read. The usual syntax for referencing regular drivers
+is something like this:
+
+```
+destination whatever {
+    my-destination(option1(value) option2(value));
+};
+```
+
+To make the syntax more native, you can use syslog-ng's block functionality
+to wrap your Python driver, hide that it's actually Python and provide a
+syntax to users of your code that is more convinient to use.
+
+```
+block destination my-destination(option1(value)
+                                 option2(value)) {
+    python(class(mydestination.MyDestination)
+                 options(option1 => `option1`,
+                         option2 => `option2`));
+
+
+};
+```
+
+This block allows the use of the more "native" syntax, completely hiding the
+fact that the implementation is Python based, concentrating on
+functionality.
+
+You should add this wrapper to your Python module in an "scl" subdirectory
+as a file with a .conf extension. syslog-ng will automatically include these
+files along the rest of the SCL.
+
+## Adding the code to syslog-ng
+
+To add Python based modules to syslog-ng, create a Python package (e.g.  a
+directory with __init__.py and anything that file references).  Add these
+files to `modules/python-modules/<subdirectory>` and open a pull request.
+
+With that a "make install" command should install your module along the
+rest of the syslog-ng binaries.
+
+You will also need to add your files to the source tarball by listing them
+in the EXTRA_DIST variable of the `modules/python-modules/Makefile.am` file.
+
+## External dependencies
+
+If your Python code depends on third party libraries, those need to be
+installed on the system where your code is deployed.
+
+If your deployment mechanism is based on packages (deb or rpm), make sure
+that you add these OS level dependencies to the packages generated.
+
+In deb this usually means adding the right package to your Depends line.
+
+In rpm this means adding the right package as a Requires line in the .spec
+file.
+
+If you would like to use pip/requirements.txt to deploy dependencies, you
+can choose to: invoke pip at "make install" time so that syslog-ng's private
+Python directory would contain all the dependencies that you require.
+
+## Adding Python code to syslog-ng deb package
+
+To add your module to the syslog-ng deb package, create a new file in
+packaging/debian/ with a name like `syslog-ng-mod-<yourmodule>.install`.
```

Comment:
[optional]

Other paths in this readme have been backticked, we could backtick this also.
```suggestion
`packaging/debian/` with a name like `syslog-ng-mod-<yourmodule>.install`.
```